### PR TITLE
feat(bigquery/storage/managedwriter): support default value controls

### PR DESCRIPTION
**Pull request by shollyman**
This PR adds new options to control how missing values are interpreted when writing.

For ManagedStream instantiation, the options are:
* `WithDefaultMissingValueInterpretation` (blanket setting for all columns)
* `WithMissingValueInterpretations` (per-column settings)

To support updates, these are added as AppendOption options:
* `UpdateDefaultMissingValueInterpretation`
* `UpdateMissingValueInterpretations`

Implementation-wise, this PR removes the previous schema-specific
versioner (`descriptorVersion`) and expands the concept to a versioned
AppendRowsRequest template (`versionedTemplate`).  This more general
mechanism allows us to version all settings that manifest as request fields 
in the AppendRowsRequest.

Fixes: https://github.com/googleapis/google-cloud-go/issues/8387

### CODE REVIEW

File: bigquery/storage/managedwriter/options.go [hunk 1]
```diff
@@ -234,7 +235,48 @@ func WithTraceID(traceID string) WriterOption {
...
+
+// WithMissingValueInterpretations controls over how missing values are interpreted
+// for individual columns.
+//
+// You must provide a map to indicate how to interpret missing value for some fields. Missing
+// values are fields present in user schema but missing in rows. The key is
+// the field name. The value is the interpretation of missing values for the
+// field.
+//
+// For example, a map {'foo': NULL_VALUE, 'bar': DEFAULT_VALUE} means all
```
Comment (quartzmo) [inline bigquery/storage/managedwriter/options.go:250]: What do you think about an example showing a map in Go with `storagepb.AppendRowsRequest_MissingValueInterpretation` values, either in place of the generic map on this line or in a Go testable example?

File: bigquery/storage/managedwriter/options.go [hunk 2]
```diff
@@ -234,7 +235,48 @@ func WithTraceID(traceID string) WriterOption {
...
+// missing values in field foo are interpreted as NULL, all missing values in
+// field bar are interpreted as the default value of field bar in table
+// schema.
+//
+// If a field is not in this map and has missing values, the missing values
+// in this field are interpreted as NULL.
```
Comment (quartzmo) [inline bigquery/storage/managedwriter/options.go:256]: Maybe mention also: "unless a non-null default was set using `WithDefaultMissingValueInterpretation`" ?

File: bigquery/storage/managedwriter/options.go [hunk 3]
```diff
@@ -278,8 +320,37 @@ type AppendOption func(*pendingWrite)
...
+// UpdateMissingValueInterpretations updates the per-column missing-value intepretations settings,
+// and is retained for subsequent writes.  See the WithMissingValueInterpretations WriterOption for
+// more details.
+func UpdateMissingValueInterpretations(mvi map[string]storagepb.AppendRowsRequest_MissingValueInterpretation) AppendOption {
+	return func(pw *pendingWrite) {
+		prev := pw.reqTmpl
+		if prev == nil {
+			prev = newVersionedTemplate()
+		}
+		pw.reqTmpl = prev.revise(reviseMissingValueInterpretations(mvi))
```
Comment (quartzmo) [inline bigquery/storage/managedwriter/options.go:340]: Why does this require two calls? Why doesn't `newVersionedTemplate` accept `...templateRevisionF` like `revise`?

Comment (shollyman) [reply bigquery/storage/managedwriter/options.go]: changed it to allowed revise() to recover from a nil start state, which simplifies the options.  We don't know based on the call order if we need to instantiate, which is why passing them into the newVersionedTemplate constructor doesn't get us as much.

File: bigquery/storage/managedwriter/options.go (hunk 2 above)
Comment (shollyman) [reply bigquery/storage/managedwriter/options.go]: updated the docstring here

File: bigquery/storage/managedwriter/options.go (hunk 1 above)
Comment (shollyman) [reply bigquery/storage/managedwriter/options.go]: inlined a more language-appropriate example of specifying the option

File: bigquery/storage/managedwriter/options.go [hunk 4]
```diff
@@ -234,7 +235,48 @@ func WithTraceID(traceID string) WriterOption {
 // AppendRows calls on the stream.
 func WithSchemaDescriptor(dp *descriptorpb.DescriptorProto) WriterOption {
 	return func(ms *ManagedStream) {
-		ms.curDescVersion = newDescriptorVersion(dp)
+		ms.curTemplate = ms.curTemplate.revise(reviseProtoSchema(dp))
+	}
+}
+
+// WithMissingValueInterpretations controls over how missing values are interpreted
```
[the remark this answers is not in the document]
Comment (shollyman) [reply bigquery/storage/managedwriter/options.go]: s/over//

**Review by alvarowolfx — APPROVED** [review]
added minor nit on missing comment and while I'm not a huge fan of the naming of the `reviseXXX` functions, I don't have a better alternative ( and they are all internal so 🤷 / and this can also be a matter of it not making much sense for a non native english speaking person haha )

File: bigquery/storage/managedwriter/integration_test.go [hunk 5]
```diff
@@ -1262,6 +1264,97 @@ func testSchemaEvolution(ctx context.Context, t *testing.T, mwClient *Client, bq
...
 }
 
+func testDefaultValueHandling(ctx context.Context, t *testing.T, mwClient *Client, bqClient *bigquery.Client, dataset *bigquery.Dataset, opts ...WriterOption) {
+	testTable := dataset.Table(tableIDs.New())
+	if err := testTable.Create(ctx, &bigquery.TableMetadata{Schema: testdata.DefaultValueSchema}); err != nil {
+		t.Fatalf("failed to create test table %s: %v", testTable.FullyQualifiedName(), err)
+	}
+
+	m := &testdata.DefaultValuesPartialSchema{
+		// We only popu
```
[the remark this answers is not in the document]
Comment (shollyman) [reply bigquery/storage/managedwriter/integration_test.go]: indeed.  completed comment.

**Pull request merged**